src/helpers/iconpicker.class.js

```diff
@@ -0,0 +1,173 @@
+                        src: imageData.content.thumbnail,
+                        copyright: imageData.content.copyright,
+                        id: imageData.content.id
+                    });
+                });
+            });
+
+            uploadPromise.catch(function(err){
+                console.log(err);
+            });
+        }
+    });
+}
+
+function createArrayOfIcons(icons) {
+    var iconsArray = [];
+    Object.keys(icons).forEach(function(k) {
+        var file = icons[k].file.replace('original', '90x90');
+        var single = _.template('<img data-id=<%= id %> src="<%= icon %>" alt="<%= alt %>" >', { icon: file, alt: icons[k].legend, id: icons[k].id });
+        iconsArray.push(single);
+    });
+    return iconsArray;
+}
+
+function getIcons(url, block) {
+    xhr.get(url)
+    .then(function(iconData) {
+        var iconsArray = createArrayOfIcons(iconData.content);
+
+        var modalInner = block.modal.$elem.children('.modal-content')[0];
+
+        if (block.modal.$elem.children('.icon-picker') === false) {
+            block.iconPicker = iconPickerHtml;
+            $(modalInner).append(block.iconPicker);
+            block.modal.$elem.children('.icon-picker').addClass('is-visible');
+        }
+        dropEvent(block);
+
+        block.modal.open();
+
+        if (block.slider === undefined) {
+            var params = {
+                contents: iconsArray,
+                itemsPerSlide: 5,
+                increment: 1,
+                container: $(modalInner),
+                controls: {
+                    next: 'Next',
+                    prev: 'Prev'
+                }
+            };
+
+            block.slider = new Slider(params); //@TODO  Teach the slider how to handle native element & jquery elements;
+        }
+
+        bindClickOnIcons(block);
+    })
+    .catch(function(err) {
+        console.log(err);
+        console.error('Somehting went wrong');
+    });
+}
+
+var IconPicker = function(param) {
+    this.apiUrl = param.apiUrl;
+    this.blockRef = param.blockRef;
+    this.modalTriggerElement = param.modalTriggerElement;
+    this.modal = new Modal({
+        slug: 'icons-modal',
+        animation: 'fade',
+        theme: 'media'
+    });
+    this.init();
+};
+
+var prototype = {
+
+    init: function(){
+        var self = this;
+
+        this.modal.render({
+            header: '<header>' + i18n.t('blocks:illustrated:modal:header') + '</header>',
+
+            content: '',
+
+            footer: {
+                ok: i18n.t('blocks:illustrated:modal:close')
+            }
+        });
+
+        if (this.modalTriggerElement.children().length !== 0){
+            this.modalTriggerElement.on('click', 'img', function() {
+                getIcons(self.apiUrl, self);
+            });
+        }
+        else {
+            this.modalTriggerElement.on('click', function() {
+                getIcons(self.apiUrl, self);
+                $(this).off('click');
```

Comment:
@arthur-balagne if you need to immediately remove an event listener after use, there is the `.one()` method